### PR TITLE
Intermedia node aggregation

### LINKED ISSUE #352: Enable int node aggregating
- [ ] Create option to aggregate intermedia node table by node attribute
- [ ] Update table on click include all aggregated nodes

### PR DESCRIPTION
**Pull request by gotdairyya**
### Does this PR close any open issues?
Closes #352 

### Give a longer description of what this PR addresses and why it's needed
This addition allows the user to aggregate the intermediate node table by node variable.

### Are there any additional TODOs before this PR is ready to go?

TODOs:
- [ ] ...

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -65,6 +65,8 @@ const {
     connectivityMatrixPaths: { nodes: [], paths: [] },
     selectedConnectivityPaths: [],
     showPathTable: false,
+    maxIntConnections: -1,
+    intAggregatedBy: 'none',
```
Comment (JackWilb) [inline src/store/index.ts:69]: I would guess this might be better as 0 and empty

File: src/components/PathTable.vue [hunk 2]
```diff
@@ -101,9 +101,9 @@ export default defineComponent({
   setup() {
     const search = ref('');
     const pathLength = computed(() => {
-      // Removes extra edge in even paths
+      // Add extra node in even paths
       if (store.state.selectedConnectivityPaths[0].vertices.length % 2 === 0) {
-        return store.state.selectedConnectivityPaths[0].vertices.length - 1;
+        return store.state.selectedConnectivityPaths[0].vertices.length + 1;
```
Comment (JackWilb) [inline src/components/PathTable.vue:106]: Not sure I have a network to test this on. Why did it need the change?

File: src/components/Controls.vue [hunk 3]
```diff
@@ -60,6 +60,19 @@ export default defineComponent({
...
+    const intAggregatedBy = computed({
+      get() {
+        return store.state.intAggregatedBy;
+      },
+      set(value: string) {
+        store.commit.setIntAggregatedBy(value);
+      },
+    });
+    const maxIntConnections = computed(() => store.state.maxIntConnections);
+    const intTableColorScale = computed(() => store.getters.intTableColorScale);
```
Comment (JackWilb) [inline src/components/Controls.vue:74]: `intTableColorScale` seems unused here

File: src/components/Controls.vue [hunk 4]
```diff
@@ -142,6 +163,10 @@ export default defineComponent({
       parentColorScale,
       nodeVariableItems,
       maxConnections,
+      showIntNodeVis,
+      intAggregatedBy,
+      maxIntConnections,
+      intTableColorScale,
```
Comment (JackWilb) [inline src/components/Controls.vue:169]: `intTableColorScale` not used in the template so can be deleted here

File: src/components/IntermediaryNodes.vue [hunk 5]
```diff
@@ -82,15 +90,44 @@ export default defineComponent({
...
-      //   Update opacity
-      const allPaths = matrix.map((row) => row.map((cell) => cell.z)).flat();
-      const maxPath = allPaths.reduce((a, b) => Math.max(a, b));
-      opacity.domain([
-        0,
-        maxPath,
-      ]);
+        //   Update opacity
+        const allPaths = matrix.map((row) => row.map((cell) => cell.z)).flat();
+        const maxPath = allPaths.reduce((a, b) => Math.max(a, b));
```
Comment (JackWilb) [inline src/components/IntermediaryNodes.vue:96]: If this is just getting the max, it's easier to read `Math.max(...allPaths)`

**Review by JackWilb — APPROVED** [review]
After looking it over, it's almost ready to go. I'll approve and let you fix the small things I found before merge

File: src/components/PathTable.vue (hunk 2 above)
Comment (gotdairyya) [reply src/components/PathTable.vue:106]: Yeah I'm not sure why my logic was wrong when I first wrote this, but the current logic excluded the last node.

This is a screen shot from the current app, notice there isn't Node 4 column
image

**Comment by gotdairyya** [comment]
I made all the small changes and see my comment about the logic change for the path table display

**Pull request merged**